subsys/net/ip/l2/ethernet/ethernet_mgmt.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_NET_DEBUG_L2_ETHERNET)
+#define SYS_LOG_DOMAIN "net/ethernet"
+#define NET_LOG_ENABLED 1
+#endif
+
+#include <errno.h>
+
+#include <net/net_core.h>
+#include <net/net_if.h>
+#include <net/ethernet_mgmt.h>
+
+static inline bool is_hw_caps_supported(struct device *dev,
+					enum ethernet_hw_caps caps)
+{
+	const struct ethernet_api *api = dev->driver_api;
+
+	return !!(api->get_capabilities(dev) & caps);
+}
+
+static int ethernet_set_config(u32_t mgmt_request,
+			       struct net_if *iface,
+			       void *data, size_t len)
+{
+	struct ethernet_req_params *params = (struct ethernet_req_params *)data;
+	struct device *dev = net_if_get_device(iface);
+	const struct ethernet_api *api = dev->driver_api;
+	enum ethernet_config_type type;
+	struct ethernet_config config;
+
+	if (!api->set_config) {
+		return -ENOTSUP;
+	}
+
+	if (!data || (len != sizeof(struct ethernet_req_params))) {
+		return -EINVAL;
+	}
+
+	if (mgmt_request == NET_REQUEST_ETHERNET_SET_AUTO_NEGOTIATION) {
+		if (!is_hw_caps_supported(dev,
+					  ETHERNET_AUTO_NEGOTIATION_SET)) {
+			return -ENOTSUP;
+		}
+
+		config.auto_negociation = params->auto_negotiation;
```

Comment:
One typo still left, or is github showing me old stuff?

Reply:
missed that one